File: my_py_pkg/my_py_pkg/add_two_ints_client_oop.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts
from functools import partial

class AddTwoIntsClient(Node):  # MODIFY NAME
    def __init__(self):
        super().__init__("add_two_ints_client_oop")  # MODIFY NAME
        self.client_ = self.create_client(AddTwoInts, "add_two_inits")

    def call_add_two_ints(self,a,b): 
        while not self.client_.wait_for_service(1.0): 
            self.get_logger().warn("waiting for add two ints server......")

        request = AddTwoInts.Request()
        request.a = 3
        request.b = 8 

        future = self.client_.call_async(request)
        future.add_done_callback(partial(self.callback_call_add_two_ints, request = request))
        
    def callback_call_add_two_ints(self, future, request): 
        response = future.result() 
        self.get_logger().info("Got Response: " + str(response.sum))
        self.get_logger().info(str(request.a) + "+" + str(request.b) + " = " + str(response.sum))



def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsClient()  
    node.call_add_two_ints(2,7)
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/add_two_ints_client_oop.cpp
    #include "rclcpp/rclcpp.hpp"
    #include "example_interfaces/srv/add_two_ints.hpp"
     
    class AddTwoIntsClientNode : public rclcpp::Node 
    {
    private: 
        rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedPtr client_ ; 
    public:
        AddTwoIntsClientNode() : Node("add_two_ints_client") 
        {
           client_ = this->create_client<example_interfaces::srv::AddTwoInts>("add_two_inits");
        }
        void callAddTwoInts(int a, int b) { 
            while(!client_->wait_for_service(std::chrono::seconds(1))){
                RCLCPP_WARN(this->get_logger(), "Waiting for the server"); 
            }
            auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
            request->a = a; 
            request->b  = b;

            client_->async_send_request(request,std::bind(&AddTwoIntsClientNode::callbackCallAddTwoInts, this, std::placeholders::_1));
        }

    private:
        void callbackCallAddTwoInts(rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedFuture future){ 
            auto response = future.get();
            RCLCPP_INFO(this->get_logger(), "Sum: %d",(int)response->sum); 
        }
    };
     
    int main(int argc, char **argv)
    {
        rclcpp::init(argc, argv);
        auto node = std::make_shared<AddTwoIntsClientNode>(); // MODIFY NAME
        node->callAddTwoInts(10,5); 
        rclcpp::spin(node);
        rclcpp::shutdown();
        return 0;
    }